Add support for individual plugin download with `tanzu plugin download-bundle` command (#725)
vmware-tanzu/tanzu-cli: docs/quickstart/install.md

tanzu plugin download-bundle --group vmware-tkg/default --to-tar /tmp/plugin_bundle_tkg_latest.tar.gz
```

If you want to download a specific plugin, the `--plugin` flag can be used. Using
the `--plugin` flag is for cases where a plugin is not part of a plugin group,

> but that normally using `--group` is the recommended approach.

Reviewer: but that normally using -> however, when possible, using

Below are the supported formats for the `--plugin` flag:

```sh
--plugin name                 : Downloads the latest available version of the plugin for all matching targets.

> --plugin name:version         : Downloads the specified version of the plugin for all matching targets. Use 'latest' as version for latest available version

Reviewer: For the two above, we no longer download all targets, right?

--plugin name@target:version  : Downloads the specified version of the plugin for the specified target. Use 'latest' as version for latest available version
--plugin name@target          : Downloads the latest available version of the plugin for the specified target.
```

```sh

> # To download plugin bundle with all available versions 'cluster' plugin across all targets

Reviewer: no longer "all available versions"

tanzu plugin download-bundle --plugin cluster --to-tar /tmp/plugin_bundle_cluster.tar.gz

> # To download plugin bundle with all available versions 'cluster' plugin for `kubernetes` target

Reviewer: same

tanzu plugin download-bundle --plugin cluster@kubernetes --to-tar /tmp/plugin_bundle_cluster.tar.gz

# To download plugin bundle with v1.0.0 version of 'cluster' plugin for `kubernetes` target
tanzu plugin download-bundle --plugin cluster@kubernetes:v1.0.0 --to-tar /tmp/plugin_bundle_cluster.tar.gz

# To download plugin bundle with latest available version of 'cluster' plugin for `kubernetes` target
tanzu plugin download-bundle --plugin cluster@kubernetes:latest --to-tar /tmp/plugin_bundle_cluster.tar.gz
```

> Using `--group` and `--plugin` flag together are also supported and if specified the union of all the

Reviewer: ```suggestion
Using the `--group` and `--plugin` flags together is also supported.  In such a case the union of all the
```
